### ISSUE TITLE
Bug in toEnum

### ISSUE THREAD

**Issue by wyager** [msakai/finite-field#2]
```
> toEnum 7 :: P.PrimeField (2^127 - 1)
*** Exception: PrimeField.toEnum: bad argument
```

`toEnum` should not use `toInt`. Instead it should convert *both* the input `Int` *and* the input `PrimeField p` to integers and compare those.

**Comment by msakai** [comment]
Good catch. Thanks!

**Comment by msakai** [comment]
@wyager, I released version `0.10.0` with the fix of this issue.
